tic-tac-toe/src/Board.jsx

```diff
@@ -0,0 +1,70 @@
+import React, {useContext} from 'react';
+import Square from './Square';
+import BoardState from './contexts/BoardContext';
+import declareWinner from './contexts/declareWinner';
+
+const Board = (props) => {
+  const { turn, setTurn, boardState, setBoardState, history, setHistory} = useContext(BoardState);
+
+  const winner = declareWinner(boardState);
+  let status;
+  if (winner) {
+    status = `The winner is ${winner}!`;
+  }
+
+  function handleClick(num) {
+    if (!declareWinner(boardState) && !boardState[num]) {
+      const updatedHistory = [...history];
+      const updatedBoard = [...boardState];
+      updatedBoard[num] = turn;
+      updatedHistory.push(updatedBoard);
+      setBoardState(updatedBoard);
+      setHistory(updatedHistory);
+      setTurn((turn === 'O') ? 'X' : 'O');
+    }
+  }
+
+  function stepTo(step) {
+    const boardInHistory = [...history];
+    setBoardState(boardInHistory[step - 1] || []);
+    setHistory(boardInHistory[step - 1] || []);
```

Comment:
It looks like we are re-setting the history each time we step backwards, so there is no way to step back forward again in the history. This means that this functions more like an undo function instead of a time-travel. For time-travel, you'd just want to not reset the history when moving backwards.